compiler/src/codegen.rs

```diff
@@ -1,4 +1,5 @@
 mod constant;
+mod constant_enum;
```

Comment:
The whole `constant_enum` is just another version of `constant` but I think the output is easier to use.
For Example for `version INTEGER {v3(3)}(v3,...),` it produces a `Version` struct with `V3` field, So you can easily write `Version::V3` wherever you need the `3`.